examples/jsm/renderers/nodes/accessors/MaterialNode.js

```diff
@@ -0,0 +1,107 @@
+import Node from '../core/Node.js';
+import OperatorNode from '../math/OperatorNode.js';
+import MaterialReferenceNode from './MaterialReferenceNode.js';
+
+class MaterialNode extends Node {
+
+	static COLOR = 'color';
+	static OPACITY = 'opacity';
+	static SPECULAR = 'specular';
+	static SHININESS = 'shininess';
+
+	constructor( scope = MaterialNode.COLOR ) {
+
+		super();
+
+		this.scope = scope;
+
+	}
+
+	getType( builder ) {
+
+		const scope = this.scope;
+		const material = builder.getContextParameter( 'material' );
+
+		if ( scope === MaterialNode.COLOR ) {
+
+			return material.map !== null ? 'vec4' : 'vec3';
+
+		} else if ( scope === MaterialNode.OPACITY ) {
+
+			return 'float';
+
+		} else if ( scope === MaterialNode.SPECULAR ) {
+
+			return 'vec3';
+
+		} else if ( scope === MaterialNode.SHININESS ) {
+
+			return 'float';
+
+		}
+
+	}
+
+	generate( builder, output ) {
+
+		const type = this.getType( builder );
+		const material = builder.getContextParameter( 'material' );
+		const scope = this.scope;
+
+		let node = null;
+
+		if (scope === MaterialNode.COLOR) {
+
+			const colorNode = new MaterialReferenceNode( 'color', 'color' );
+
+			if (material.map !== null && material.map !== undefined) {
```

Comment:
How about?
```js
if ( material.isTexture === true ) {
```

Reply:
I think that `!==` follows the pattern of others... `alphaMap`, `specularMap`, etc.

Reply:
Do you want me to change? Or do we continue like `!==` for now?

Reply:
I've meant to change it everywhere in `MaterialNode`. In this case, you can ensure that the code actually works with a texture and not something else.

Reply:
Ahh, yeah, now I understand,... Did you mean `texture.isTexture`, right?
```js
if ( texture.isTexture === true ) {
```